### PR TITLE
Die shardalias die

### PR DESCRIPTION
**Pull request by robin900**
Remove all usage of shardalias in code and tests.

Fixes citusdata/citus#733.

### CODE REVIEW

File: src/backend/distributed/citus.sql [hunk 1]
```diff
@@ -42,7 +42,6 @@ CREATE TABLE citus.pg_dist_shard(
     logicalrelid oid NOT NULL,
     shardid int8 NOT NULL,
     shardstorage "char" NOT NULL,
-    shardalias text,
```
Comment (marcocitus) [inline src/backend/distributed/citus.sql]: Given that the alter is now gone, this guy and the headers should probably remain.

Comment (robin900) [inline src/backend/distributed/citus.sql]: My understanding is that citus.sql is for new installs of the extension. And regression tests...seem to create a brand new database and extension install on each run.

So, we would want:

1. new installs of citus extension to omit shardalias;
2. regression tests to omit shardalias in DDL (to guard against code referencing the column);
3. allow existing citus installs to keep shardalias column.

Did I get all that right?

File: src/include/distributed/pg_dist_shard.h [hunk 2]
```diff
@@ -43,13 +42,12 @@ typedef FormData_pg_dist_shard *Form_pg_dist_shard;
  *      compiler constants for pg_dist_shards
  * ----------------
  */
-#define Natts_pg_dist_shard 6
```
Comment (robin900) [inline src/include/distributed/pg_dist_shard.h]: @marcocitus are these changes incompatible if the pg_dist_shard table retains the shardalias column? It seems that they are incompatible. So how about I do the following:

1. Preserve the presence of shardalias column in citus.sql and in relevant regression tests.
2. Revert the changes in pg_dist_shard.h, but  change Anum_pg_dist_shard_shardalias to Anum_pg_dist_shard_shardalias_DEPRECATED.
3. Restore the code in master_metadata_utility.c that sets null on shardalias in the constructed tuple.

Does that sound right?

**Comment by marcocitus** [comment]
Looks good to me :). Thanks for doing this! Could you collapse the changes into 1 commit?

**Comment by robin900** [comment]
@marcocitus done. travis running now.

**Comment by lithp** [comment]
@jasonmp85 Any chance you could look at this PR please? It's a :shipit: but the Merge Pull Request button is greyed out for me. My theory is that since this branch isn't in the citus repo travis never gets to it and toggles the status check. I'm also not allowed to merge and push from the command line since I'm not an admin.

File: src/backend/distributed/master/master_stage_protocol.c [hunk 3]
```diff
@@ -493,19 +489,15 @@ UpdateShardStatistics(int64 shardId)
 	text *minValue = NULL;
 	text *maxValue = NULL;
 
-	/* if shard doesn't have an alias, extend regular table name */
-	shardQualifiedName = LoadShardAlias(relationId, shardId);
-	if (shardQualifiedName == NULL)
-	{
-		char *shardName = get_rel_name(relationId);
+	/* Build shard qualified name. */
+	char *shardName = get_rel_name(relationId);
```
Comment (lithp) [inline src/backend/distributed/master/master_stage_protocol.c]: Unnecessary blank line

**Comment by lithp** [comment]
@robin900 Thanks for the PR, I tried my hardest to find a way to contribute but this is close to perfect! 

Just a few things:
- There's an unnecessary blank line in `master_stage_protocol.c:494`
- Both `multi_physical_planner` and `master_delete_protocol` no longer need to include `master_metadata_utility`
- The addition to `.gitignore` is an unrelated change which we generally frown upon

If you get around to it before Jason takes a look, one more commit and a squash would be appreciated, but none of these are big enough to block the PR. This is a nice change, thanks again for submitting it! You're just a merge away from being the only person to have contributed negative code to Citus ;)

**Comment by robin900** [comment]
@lithp @jasonmp85 Amended commit to include your suggestions, ran tests successfully, force pushed to my repo, Travis build passed.

**Comment by robin900** [comment]
Rebased to citus/master and force pushed; once Travis build succeeds again, you should be clear to merge.

**Comment by lithp** [comment]
Merged :tada:

**Comment by jasonmp85** [comment]
(FWIW the issue was that the branch was "out of date" with the main branch. It was the rebase that fixed the problem).

**Pull request merged**